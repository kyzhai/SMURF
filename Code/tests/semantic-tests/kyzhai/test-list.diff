61a62,63
> 	ID: head h : t :: [a] -> a
> 	ID: tail h : t :: [a] -> [a]
