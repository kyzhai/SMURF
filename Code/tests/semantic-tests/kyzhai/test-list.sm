a = [True, False, True, False]

main = []
