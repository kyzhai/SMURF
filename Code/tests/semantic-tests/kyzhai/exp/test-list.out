Program: 
	SVardef: 
		ID: a  :: []

	[]
	SVardef: 
		ID: b  :: [[]]

	[[]]
	SVardef: 
		ID: c  :: []

	[] ++ []
	SVardef: 
		ID: f  :: [[]]

	[[]] ++ []
	SVardef: 
		ID: d  :: [Int]

	[1] ++ [2]
	SVardef: 
		ID: g  :: [Int]

	[] ++ [1, 2, 3, 4]
	SVardef: 
		ID: h  :: [[]]

	[] : []
	SVardef: 
		ID: i  :: [Int]

	1 : []
	SVardef: 
		ID: j  :: [[Int]]

	[1, 2] : [[1], [2, 3]]
	SVardef: 
		ID: aa  :: Chord

	[ (1, 2)$2..] ++ []
	SVardef: 
		ID: bb  :: System

	[ (1, 2)$2..] : []
	SMain: []
Global Scope: 
	ID: main  :: []
	ID: a  :: []
	ID: b  :: [[]]
	ID: c  :: []
	ID: f  :: [[]]
	ID: d  :: [Int]
	ID: g  :: [Int]
	ID: h  :: [[]]
	ID: i  :: [Int]
	ID: j  :: [[Int]]
	ID: aa  :: Chord
	ID: bb  :: System
	ID: print x :: a -> a
	ID: random  :: Int
	ID: head h : t :: [a] -> a
	ID: tail h : t :: [a] -> [a]

